README.md
# storybuilderunite
A develop repository for new story builder


CHANGELOG.md
# ChangeLog
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).

## [Unreleased]

## [0.6.2-0] - 2021-01-29
### Changed
- new builder

## [0.5.1-4] - 2020-02-26
### Fixed
- rubi: kotatsu

## [0.5.1-3] - 2020-02-18
### Fixed
- rubi

## [0.5.1-2] - 2020-02-18
### Fixed
- format invalid breakline

## [0.5.1-1] - 2020-02-01
### Added
- episode char count in description

## [0.5.1] - 2020-02-01
### Added
- description output with scene char count

## [0.5.0] - 2020-02-01
### Added
- continued action

## [0.4.5-7] - 2020-01-31
### Changed
- person line format

## [0.4.5-6] - 2020-01-31
### Added
- person line output
### Changed
- stage and event line output dir

## [0.4.5-5] - 2020-01-31
### Fixed
- event point string longer

## [0.4.5-4] - 2020-01-31
### Fixed
- event line scene title

## [0.4.5-3] - 2020-01-31
### Changed
- event line format
### Fixed
- stage and area cooperation

## [0.4.5-2] - 2020-01-30
### Fixed
- Asset file names

## [0.4.5-1] - 2020-01-30
### Fixed
- Area not set bug

## [0.4.5] - 2020-01-30
### Added
- Area data class

## [0.4.4-5] - 2020-01-29
### Added
- elapsed day

## [0.4.4-4] - 2020-01-29
### Added
- Asset: acccessory
### Changed
- block output using list switch
### Fixed
- time delta in Stageline

## [0.4.4-3] - 2020-01-28
### Fixed
- duplicated maru in Conte

## [0.4.4-2] - 2020-01-28
### Added
- Time object new attr and methods: elapsed time

## [0.4.4-1] - 2020-01-28
### Fixed
- desc char count: except wear action strings

## [0.4.4] - 2020-01-28
### Changed
- object texture: single data from dict

## [0.4.3-33] - 2020-01-28
### Added
- scene objects output on Conte

## [0.4.3-32] - 2020-01-28
### Changed
- block info output using tags
### Fixed
- action object dupulicated when dividing actions

## [0.4.3-31] - 2020-01-28
### Changed
- stage line format

## [0.4.3-30] - 2020-01-27
### Fixed
- question and kakko bug: invalid inserting a space

## [0.4.3-29] - 2020-01-27
### Changed
- look action and subject display format on conte

## [0.4.3-28] - 2020-01-27
### Fixed
- combine to description bug: divided action

## [0.4.3-27] - 2020-01-27
### Changed
- stage line output format: time delta arrow

## [0.4.3-26] - 2020-01-27
### Added
- info volume output

## [0.4.3-25] - 2020-01-26
### Added
- event line output
### Changed
- stage line format

## [0.4.3-24] - 2020-01-26
### Added
- meta data: event

## [0.4.3-23] - 2020-01-26
### Fixed
- info tag convert

## [0.4.3-22] - 2020-01-25
### Added
- block title display on conte

## [0.4.3-21] - 2020-01-25
### Fixed
- part output end number

## [0.4.3-20] - 2020-01-24
### Fixed
- part output number

## [0.4.3-19] - 2020-01-24
### Added
- action per person output

## [0.4.3-18] - 2020-01-23
### Fixed
- conte output in action do with item object

## [0.4.3-17] - 2020-01-23
### Fixed
- kigou with spacing bug: invalid inserted

## [0.4.3-16] - 2020-01-21
### Fixed
- who bug: invalid replacement

## [0.4.3-15] - 2020-01-21
### Added
- auto action divide

## [0.4.3-14] - 2020-01-21
### Fixed
- mecab verbs analyzer

## [0.4.3-13] - 2020-01-21
### Fixed
- writer subject using not person

## [0.4.3-12] - 2020-01-21
### Changed
- conte output format (icon adding)

## [0.4.3-11] - 2020-01-20
### Added
- chapter and episode char count output
- block info output
- stage line output
### Changed
- scene setting shared over episodes
### Fixed
- comment output always

## [0.4.3-10] - 2020-01-15
### Added
- note tag convert
- charactor count output each scene on conte

## [0.4.3-9] - 2020-01-13
### Added
- text output

## [0.4.3-8] - 2020-01-13
### Fixed
- tag format: br and symbol output

## [0.4.3-7] - 2020-01-12
### Added
- columns and rows setting

## [0.4.3-6] - 2020-01-12
### Fixed
- combine documents using then

## [0.4.3-5] - 2020-01-12
### Added
- set outline in world
### Fixed
- output outline maintile

## [0.4.3-4] - 2020-01-12
### Added
- note and titles counter
### Changed
- outline output format (each heads)

## [0.4.3-3] - 2020-01-10
### Fixed
- buildDB order typo

## [0.4.3-2] - 2020-01-10
### Fixed
- chapter count ouput: typo

## [0.4.3-1] - 2020-01-10
### Fixed
- output without maru using ActType.Voice

## [0.4.3] - 2020-01-08
### Added
- LifeNote
- History

## [0.4.2] - 2020-01-07
### Added
- MetaData
- Rubi
- Layer
- Pronoun
- Checker
### Removed
- Flag

## NOTE
- version 0.4.0 and 0.4.1 is fallback

## [0.3.2] - 2019-12-16
### Added
- Writer
- Scene: using when, where and who
### Changed
- Analyzer: using story containers
### Fixed
- old parser methods

## [0.3.1] - 2019-12-02
### Added
- Extractor
- Formatter
- Covnerter
- common times data
- episode char count
- Person: simple person creator
- test utility
### Changed
- Parser
- display names of values in subtest
- Analyzer: containsWord using all story containers/And or Or check enable
### Fixed
- analyzer bug
- default layer

## [0.3.0] - 2019-11-17
### Added
- action layer
- using MeCab for analyzer
- priority filter
- collect word class
- parser class
- formatter class
- test utility
- build test on travis ci
### Changed
- scene set at first time
### Fixed
- scene and episode title
- scenario symbole without maru


## [0.2.10] - 2019-11-13
### Note
- moved new repository

## [0.2.9] - 2019-10-18
### TODO
- each unit tests
### Changed
- Refining total sources
### Removed
- Old builder

## [0.2.1] - 2019-07-07
### Added
- themes and motifs checking utility
- manupapers count
- keywords checking in descriptions
- first and last name each persons (using as tag)
- person's new attributes
- emphasis description
- directly writing a description to an action
- output each scene information
### Fixed
- duplicated output with info option
- empty main title error
- nodesc class checkout

## [0.2.0] - 2019-04-27
### Added
- Analyzer methods from tools
- Parser methods from tools

example/config.py
# -*- coding: utf-8 -*-
"""Sample config for the momotaro.
"""


################################################################
#
# Sample:
#
# 1) PERSONS
#       人物簡易設定
# 2) STAGES
#       舞台基本設定
# 3) DAYS
#       年月日設定
# 4) TIMES
#       時間帯基本設定
# 5) ITEMS
#       小道具設定
# 6) WORDS
#       辞書設定
# 7) RUBIS
#       ルビ設定
# 8) LAYERS
#       レイヤ設定
#
################################################################


PERSONS = (
        # Tag / 氏,名 / 歳 / 性別 / 職業 / 呼称 / 紹介
        ("taro", "桃太郎", "", 15, "male", "無職", "me:俺"),
        ("granpa", "おじいさん", "守屋,源三郎", 75, "male", "木こり", "me:儂:S:おじいさん:granma:お前"),
        ("granma", "おばあさん", "守屋,富", 74, "female", "農家", "me:私:S:おばあさん:granpa:あなた"),
        ("daemon1", "赤鬼", "", 35, "male", "鬼", "me:ワイ"),
        ("dog", "犬助", "", 18, "male", "犬", "me:ボク"),
        ("monkey", "猿衛門", "", 16, "male", "猿", "me:オイラ"),
        ("pheasant", "雉子", "", 17, "female", "雉", "me:わたし"),
        )

STAGES = (
        # Tag / 名前 / 紹介
        ("river", "川辺"),
        ("home", "太郎の家"),
        ("vila", "村"),
        ("street", "路上"),
        ("ship", "船"),
        ("island", "鬼ヶ島", "村からは十日かかる"),
        )

DAYS = (
        # Tag / 名前 / 月 / 日 / 年
        ("getpeach", "桃拾った", 7,7, 10),
        ("birth", "太郎生まれた", 7,7, 10),
        ("rumor", "鬼の噂", 9,1, 13),
        ("voyage", "出発の日", 10,1, 13),
        ("meetdog", "犬に会う", 10,4, 13),
        ("rideship", "船にて", 10,10, 13),
        ("arrived", "鬼ヶ島に到着", 10,11, 13),
        ("backhome", "村に帰る", 10,21, 13),
        )

TIMES = (
        # Tag / 名前 / 時 / 分
        ("morning", "朝", 8, 0),
        ("midmorning", "午前", 10, 0),
        ("afternoon", "午後", 14, 0),
        ("night", "夜", 19, 0),
        )

ITEMS = (
        # Tag / 名前 / 紹介
        ("peach", "桃"),
        ("dango", "吉備団子", "家来を餌付けするための団子"),
        ("floor", "床"),
        ("ceil", "天井"),
        ("wall", "壁"),
        ("water", "水"),
        ("tree", "木"),
        ("land", "陸"),
        ("cloth_momo1", "桃太郎の服"),
        ("cloth_momo2", "桃太郎の服（外出着）"),
        ("katana", "刀"),
        ("flag", "旗"),
        ("cloth_boro", "ボロ着"),
        )

WORDS = (
        # Tag / 名前 / 紹介
        ("island", "鬼ヶ島"),
        )

RUBIS = (
        # Base / Rubi / Type
        ("桃太郎", "｜桃太郎《ももたろう》", "", 1),
        ("桃", "｜桃《モモ》", ("桃太郎", "桃太"), 2),
        ("吉備団子", "｜吉備団子《きびだんご》", "", 1),
        ("木の棒", "｜木の棒《きのぼう》", "", 1),
        ("富", "｜富《とみ》", "", 1),
        )

LAYERS = (
        ("main", "桃太郎", ("桃太郎", "桃")),
        ("buster", "鬼退治", ("鬼", "鬼退治", "退治")),
        )
